app/models/post.rb

```diff
@@ -2,4 +2,34 @@ class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  after_create :increment_user_posts_counter
+  after_destroy :decrement_user_posts_counter
+
+  after_create :update_likes_counter
+  after_destroy :update_likes_counter
+
+  private
+
+  def increment_user_posts_counter
+    user.increment!(:posts_count)
+  end
```

Comment:
- The column referenced in the `Post` model is the `author` column, so you should increment the `author` instead of user. Please fixed this 🙏 
- Also, the attribute in the User model is `posts_counter` and not `posts_count`, please fix this to avoid errors in your code 😄